core/src/main/scala/akka/persistence/r2dbc/internal/codec/TimestampCodec.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2022 - 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.r2dbc.internal.codec
+
+import java.time.Instant
+import java.time.LocalDateTime
+import java.util.TimeZone
+
+import io.r2dbc.spi.Row
+import io.r2dbc.spi.Statement
+
+import akka.annotation.InternalApi
+import akka.persistence.r2dbc.internal.InstantFactory
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] sealed trait TimestampCodec {
+
+  def encode(timestamp: Instant): Any
+  def decode(row: Row, name: String): Instant
+  def decode(row: Row, index: Int): Instant
+
+  // should we name it just `now()`? The type should not be in the name...
+  def instantNow(): Instant = InstantFactory.now()
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object TimestampCodec {
+
+  class PostgresTimestampCodec extends TimestampCodec {
+    override def decode(row: Row, name: String): Instant = row.get(name, classOf[Instant])
+    override def decode(row: Row, index: Int): Instant = row.get(index, classOf[Instant])
+
+    override def encode(timestamp: Instant): Any = timestamp
+  }
+  object PostgresTimestampCodec extends PostgresTimestampCodec
+
+  case object SqlServerTimestampCodec extends TimestampCodec {
+
+    // should this come from config?
+    private val zone = TimeZone.getTimeZone("UTC").toZoneId
```

Comment:
slightly more direct:
```
private val zone = ZoneId.of("UTC")
```

Reply:
Changed it. Do you think the string `UTC` should come from config so it can be overridden by users?

Reply:
it should not be configurable, we should always use UTC